superset-frontend/src/components/ReportModal/index.tsx

```diff
@@ -235,15 +244,19 @@ const ReportModal: FunctionComponent<ReportProps> = ({
       await dispatch(
         editReport(currentReport?.id, newReportValues as ReportObject),
       );
+      onHide();
     } else {
-      await dispatch(addReport(newReportValues as ReportObject));
+      try {
+        await dispatch(addReport(newReportValues as ReportObject));
+        onHide();
+      } catch (e) {
+        const parsedError = await getClientErrorObject(e);
+        const errorMessage = parsedError.message;
+        onReducerChange(ActionType.error, errorMessage);
+      }
     }
 
-    if (onReportAdd) {
-      onReportAdd();
-    }
-
-    onClose();
+    if (onReportAdd) onReportAdd();
```

Comment:
Is this the format preferred by prettier these days? I learned that braceless if clauses should be avoided because they can lead to subtle errors.

Reply:
It's not preferred by prettier, I thought it was just a _prettier_ way to do it, haha. I didn't realize there were subtle errors possible with this though, thanks for this catch! I'll change it back.

Reply:
I think a single line is OK, since it's simple enough. The use that is discouraged is:

```js
if (onReportAdd)
  onReportAdd();
```

Which I think `prettier` would remove.

I think it's fine to leave the way you did, I was just curious. :)

Reply:
Oh okay awesome, I'll keep an eye out for that behavior in the future just in case! Thank you! 😁